devicetypes/smartthings/zwave-water-temp-humidity-sensor.src/zwave-water-temp-humidity-sensor.groovy

```diff
@@ -0,0 +1,219 @@
+		secure(zwave.sensorMultilevelV5.sensorMultilevelGet(sensorType: 0x01)), // temperature
+		"delay 500",
+		secure(zwave.sensorMultilevelV5.sensorMultilevelGet(sensorType: 0x05)), // humidity
+		"delay 10000",
+		secure(zwave.wakeUpV2.wakeUpNoMoreInformation())
+	])
+}
+
+def updated() {
+	configure()
+}
+
+def configure() {
+	sendEvent(name: "checkInterval", value: 8 * 60 * 60 + 10 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+}
+
+def parse(String description) {
+	def results = []
+	if (description.startsWith("Err")) {
+		results += createEvent(descriptionText: description, displayed: true)
+	} else {
+		def cmd = zwave.parse(description)
+		if (cmd) {
+			results += zwaveEvent(cmd)
+		}
+	}
+	log.debug "parse() result ${results.inspect()}"
+	return results
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand()
+	if (encapsulatedCommand) {
+		zwaveEvent(encapsulatedCommand)
+	} else {
+		log.warn "Unable to extract encapsulated cmd from $cmd"
+		createEvent(descriptionText: cmd.toString())
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.notificationv3.NotificationReport cmd) {
+	if (cmd.notificationType == 0x05) {
+		if (cmd.event == 0x01 || cmd.event == 0x02) {
+			sensorWaterEvent(1)
+		} else if (cmd.event == 0x00) {
+			sensorWaterEvent(0)
+		}
+	} else if (cmd.notificationType == 0x07) {
+		if (cmd.event == 0x03) {
+			runIn(10, clearTamper, [overwrite: true, forceForLocallyExecuting: true])
+			createEvent(name: "tamper", value: "detected", descriptionText: "$device.displayName was tampered")
+		} else if (cmd.event == 0x00) {
+			createEvent(name: "tamper", value: "clear")
+		}
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sensorbinaryv2.SensorBinaryReport cmd) {
+	return sensorWaterEvent(cmd.sensorValue)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd) {
+	return sensorWaterEvent(cmd.value)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.batteryv1.BatteryReport cmd) {
+	def map = [name: "battery", unit: "%", isStateChange: true]
+	state.lastbatt = now()
+	if (cmd.batteryLevel == 0xFF) {
+		map.value = 1
+		map.descriptionText = "$device.displayName battery is low!"
+	} else {
+		map.value = cmd.batteryLevel
+	}
+	createEvent(map)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sensormultilevelv5.SensorMultilevelReport cmd) {
+	def map = [:]
+	switch (cmd.sensorType) {
+		case 0x01:
+			map.name = "temperature"
+			map.unit = temperatureScale
+			map.value = convertTemperatureIfNeeded(cmd.scaledSensorValue, cmd.scale == 1 ? "F" : "C", cmd.precision)
+			break
+		case 0x05:
+			map.name = "humidity"
+			map.value = cmd.scaledSensorValue.toInteger()
+			map.unit = "%"
+			break
+		default:
+			map.descriptionText = cmd.toString()
+	}
+	createEvent(map)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.wakeupv2.WakeUpNotification cmd) {
+	def cmds = []
```

Comment:
Please break code blocks up (not just in this function) with empty lines so that the code is easier to read. For example, an empty line after variable declarations, and empty line before and after if statements, and an empty line before return statements.